include/jemalloc/internal/rtree_types.h

```diff
@@ -25,11 +26,17 @@ typedef struct rtree_s rtree_t;
 #define RTREE_HEIGHT_MAX						\
     ((1U << (LG_SIZEOF_PTR+3)) / RTREE_BITS_PER_LEVEL)
 
+/*
+ * Number of leafkey/leaf pairs to cache.  Each entry supports an entire leaf,
+ * so the cache hit rate is typically high even with a small number of entries.
+ * Two entries is enough to keep the hit rate high even when an arena uses a
+ * combination of dss and mmap.
+ */
+#define RTREE_CTX_NCACHE 2
```

Comment:
I suggest going with 8 entries, which is 128 bytes (two cachelines). With the adjacent cacheline prefetching (always enabled I believe), getting the first cacheline always bring us the next neighbor anyway. Since even the cache miss here isn't too bad, I'm inclined to say no need going very aggressive here. 

Also worth making sure the it's cacheline aligned.